maven/bnd-resolver-maven-plugin/src/main/java/aQute/bnd/maven/resolver/plugin/ResolverMojo.java

```diff
@@ -0,0 +1,63 @@
+package aQute.bnd.maven.resolver.plugin;
+
+import aQute.bnd.build.Workspace;
+import aQute.bnd.service.RepositoryPlugin;
+
+import biz.aQute.resolve.Bndrun;
+import biz.aQute.resolve.StandaloneBndrun;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+/**
+ * Resolves the <code>-runbundles</code> for the given bndrun file.
+ */
+@Mojo(name = "resolve", requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
+public class ResolverMojo extends AbstractMojo {
+	@Parameter(readonly = true, required = true)
+	private List<File>	bndruns;
+
+	@Parameter(defaultValue = "true")
+	private boolean				failOnChanges;
+
+	@Parameter(defaultValue = "${session}", readonly = true)
+	private MavenSession session;
+
+	public void execute() throws MojoExecutionException, MojoFailureException {
+		try {
+			for (File runFile : bndruns) {
+				resolve(runFile);
+			}
+		} catch (Exception e) {
+			throw new MojoExecutionException(e.getMessage(), e);
+		}
+	}
+
+	private void resolve(File runFile) throws MojoExecutionException, Exception, IOException {
+		if (!runFile.exists()) {
+			throw new MojoExecutionException("Could not find bnd run file " + runFile);
```

Comment:
For the same reason when you compile 20 java source files, the compiler gives you errors and warnings on all the source files before ending compilation. If the compiler gave up at the first source file, it would take many compiler iterations to solve all the errors. This is build time tool and should help the developer as much as possible.

Failfast is good when you need to prevent data corruption. I don't think that is the case here :-)